CSharp/ArticleProjects/EfficientListQueries/Program.cs

```diff
@@ -0,0 +1,273 @@
+﻿// Copyright (c) Microsoft Corporation
+//
+// Companion project backing the code snippets found in the following article:
+// https://azure.microsoft.com/documentation/articles/batch-efficient-list-queries/
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Batch;
+using Microsoft.Azure.Batch.Auth;
+using Microsoft.Azure.Batch.Common;
+using Microsoft.Azure.Batch.Samples.Common;
+
+namespace Microsoft.Azure.Batch.Samples.Articles.EfficientListQueries
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            // Each task that we create below is intentionally quite large (>2048 bytes) so
+            // you may wish to adjust this value (up or down) based on your network connection.
+            const int taskCount = 5000;
+
+            const string poolId = "pool1";
+            const string jobId  = "job1";

```

Comment:
I'd prefer to have more descriptive names here. Since pools are persistent (don't go away until deleted), the user may have a number of leftover pools from other samples so tying the pool name to the sample would be best.